pkg/ingester/ingester.go

```diff
@@ -475,11 +483,15 @@ func (i *Ingester) MetricsForLabelMatchers(ctx old_ctx.Context, req *client.Metr
 
 // UserStats returns ingestion statistics for the current user.
 func (i *Ingester) UserStats(ctx old_ctx.Context, req *client.UserStatsRequest) (*client.UserStatsResponse, error) {
+	userID, err := user.ExtractOrgID(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("no user id")
+	}
 	i.userStatesMtx.RLock()
 	defer i.userStatesMtx.RUnlock()
-	state, err := i.userStates.getOrCreate(ctx)
-	if err != nil {
-		return nil, err
+	state, ok := i.userStates.get(userID)
```

Comment:
Do you mean to reduce repetition?  It comes out a bit of a wash because you still need two if-statements for error and not-found.

Otherwise what would it do with the context?

Reply:
I put in `getViaContext()`; it came out the same number of lines but is a bit more readable.